plume-models/src/users.rs

```diff
@@ -138,6 +141,14 @@ impl User {
     find_by!(users, find_by_name, username as &str, instance_id as i32);
     find_by!(users, find_by_ap_url, ap_url as &str);
 
+    pub fn is_moderator(&self) -> bool {
+        self.role == 0 || self.role == 1
+    }
+
+    pub fn is_admin(&self) -> bool {
+        self.role == 0
+    }
+
```

Comment:
can't we use an enum here instead of these numbers? (or in SQLite, a check constraint https://stackoverflow.com/questions/5299267/how-to-create-enum-type-in-sqlite)

Reply:
See this commit message https://github.com/Plume-org/Plume/pull/619/commits/e3ecbdb2f4a8aa2b857057b27cd75b44c379767b

I tried to use a crate to do that with Diesel, but it generates a schema with the enum type for Postgres and with `Text` for SQlite, which we would have to maintain two different `User` struct depending on the database, one with `role: UserRole,` and the other with `role: String,`, and I don't want to do that.

Reply:
Having an enum to make the comparison and not having magic numbers everywhere would still be cool
```rust
enum Role {
    Admin = 0,
    Mod = 1,
    User = 2,
}
fn main() {
    println!("{}", 0 == Role::Admin as u32)
}
```

Reply:
I didn't know it was possible to associate numbers to enum variants like that.